docs/kinesis/kinesis_connector.md

```diff
@@ -0,0 +1,149 @@
+ format. 
+ An example of a valid JSON string: 
+ `{"sensorId": 17,"currentTemperature": 147,"status": "WARN"}`.
+ An example of an invalid JSON format:
+ `{"sensorId": 17,"currentTemperature": 147,"status": "WARN",}`. The Exasol
+ Kinesis Connector will not parse this string correctly.
+
+## Deployment
+
+Please refer to the [deployment guide](../deployment_guide.md) till the section
+`Upload the JAR file to the bucket` (inclusive) to prepare a bucket and a JAR.
+
+## Preparing a Table for Data
+
+To store the data from a stream, we need a table. 
+The columns in the Exasol table have to imitate the data types of the data
+ stored in the stream and also be in the exact order.
+ 
+The table also requires two additional columns to store the Kinesis metadata:  
+    `KINESIS_SHARD_ID VARCHAR(2000), 
+     SHARD_SEQUENCE_NUMBER VARCHAR(2000)`
+These two columns must be at the end of the table.
+ 
+For example, given the following stream data, 
+`{"sensorId": 17,"currentTemperature": 147,"status": "WARN"}`, 
+the Exasol table should be as below:
+
+```sql
+CREATE SCHEMA <schema_name>;
+
+CREATE OR REPLACE TABLE <schema_name>.<table_name> 
+    (SENSORID DECIMAL(18,0), 
+    CURRENTTEMPERATURE DECIMAL(18,0), 
+    STATUS VARCHAR(100), 
+-- Required for importing data from Kinesis
+    KINESIS_SHARD_ID VARCHAR(130), 
+    SHARD_SEQUENCE_NUMBER VARCHAR(2000));
+
+```
+
+## Create ETL UDFs Scripts
+
+Create the following UDF scripts. Please do not change the names of the scripts.
+
+```sql
+--/
+CREATE OR REPLACE JAVA SET SCRIPT KINESIS_METADATA (...) 
+EMITS (KINESIS_SHARD_ID VARCHAR(130), SHARD_SEQUENCE_NUMBER VARCHAR(2000)) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-<VERSION>.jar;
+/
+;
+  
+--/
+CREATE OR REPLACE JAVA SET SCRIPT KINESIS_IMPORT (...) EMITS (...) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisShardDataImporter;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-<VERSION>.jar;
+/
+;
+    
+--/
+CREATE OR REPLACE JAVA SET SCRIPT KINESIS_PATH (...) EMITS (...) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisImportQueryGenerator;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-<VERSION>.jar;
+/
+; 
+```
+
+## Import Data
+
+You can provide AWS credentials using:
+1. UDF parameters 
+1. [Exasol named connection object][exa-connection]. 
+
+We highly **recommend** using connection objects to provide credentials 
+to UDFs so that credentials do not display in audit logs.
+
+### Using Connection Object
+
+First, create a named connection object and encode credentials as key-value
+pairs separated by semicolon (`;`).
+
+`AWS_SESSION_TOKEN` is an optional property. You can remove it from the string
+if you don't need it. 
+
+```sql
+CREATE OR REPLACE CONNECTION KINESIS_CONNECTION
+TO ''
+USER ''
+IDENTIFIED BY 'AWS_ACCESS_KEY=<AWS_ACCESS_KEY>;AWS_SECRET_KEY=<AWS_SECRET_KEY>;AWS_SESSION_TOKEN=<AWS_SESSION_TOKEN>';
```

Comment:
Please mention where to get those credentials briefly. E.g. by pointing to the corresponding AWS documentation.

Below is a table with all properties. Maybe just add links there.

Reply:
Added access keys as a requirement and also inserted a link